byzcoin/bcadmin/cmd_db.go

```diff
@@ -496,25 +507,36 @@ func dbCheck(c *cli.Context) error {
 					continue
 				}
 				if len(previous.ForwardLink) <= i {
-					log.Warnf("%s points to block with not enough forward"+
-						"-links", errStrBl)
 					continue
 				}
 				if previous.ForwardLink[i].IsEmpty() {
 					continue
 				}
 				if !previous.ForwardLink[i].To.Equal(sb.Hash) {
 					log.Errorf(
-						"%s pointing backwards to a block that doesn't"+
-							" reference it", errStrBl)
+						"%s pointing backwards to a block that references"+
+							" another block, which indicates a fork", errStrBl)
 				}
 			}
 		}
-		for i, fl := range sb.ForwardLink {
-			if fl.IsEmpty() {
-				continue
+
+		// Verify available forward-links to match the backward-links of the
+		// blocks they point to.
+		// If enabled, also verify the signature of the forward-links.
+		indexes := sb.GetFLIndexes()
+		for i, index := range indexes {
+			if index > last.Index {
+				break
 			}
 			errStrFl := fmt.Sprintf("%s forwardLink at height %d", errStr, i)
+
+			if i > sb.GetForwardLen() || sb.ForwardLink[i].IsEmpty() {
```

Comment:
```suggestion
			if i >= sb.GetForwardLen() || sb.ForwardLink[i].IsEmpty() {
```

In the case there is one FL in the block instead of 2 for example, in the second iteration, `1 > 1` is false, then you do sb.ForwardLink[1], which will panic.

Reply:
Thanks! I can't believe no test failed for this. Besides not believing I still do those off-by-1-errors :(